tabby-core/src/services/profiles.service.ts

```diff
@@ -211,27 +349,192 @@ export class ProfilesService {
+    * Does not return builtin groups
+    */
+    getSyncProfileGroups (): PartialProfileGroup<ProfileGroup>[] {
+        return deepClone(this.config.store.groups ?? [])
+    }
+
+    /**
+    * Return an Array of the existing ProfileGroups
+    * arg: includeProfiles (default: false) -> if false, does not fill up the profiles field of ProfileGroup
+    * arg: includeNonUserGroup (default: false) -> if false, does not add built-in and ungrouped groups
+    */
+    async getProfileGroups (includeProfiles = false, includeNonUserGroup = false): Promise<PartialProfileGroup<ProfileGroup>[]> {
+        let profiles: PartialProfile<Profile>[] = []
+        if (includeProfiles) {
+            profiles = await this.getProfiles(includeNonUserGroup, true)
+        }
+
+        let groups: PartialProfileGroup<ProfileGroup>[] = this.getSyncProfileGroups()
+        groups = groups.map(x => {
+            x.editable = true
+
+            if (includeProfiles) {
+                x.profiles = profiles.filter(p => p.group === x.id)
+                profiles = profiles.filter(p => p.group !== x.id)
+            }
+
+            return x
+        })
+
+        if (includeNonUserGroup) {
+            const builtInGroups: PartialProfileGroup<ProfileGroup>[] = []
+            builtInGroups.push({
+                id: 'built-in',
+                name: this.translate.instant('Built-in'),
+                editable: false,
+                profiles: [],
+            })
+
+            const ungrouped: PartialProfileGroup<ProfileGroup> = {
+                id: 'ungrouped',
+                name: this.translate.instant('Ungrouped'),
+                editable: false,
+            }
+
+            if (includeProfiles) {
+                for (const profile of profiles.filter(p => p.isBuiltin)) {
+                    let group: PartialProfileGroup<ProfileGroup> | undefined = builtInGroups.find(g => g.id === slugify(profile.group ?? 'built-in'))
+                    if (!group) {
+                        group = {
+                            id: `${slugify(profile.group!)}`,
+                            name: `${profile.group!}`,
+                            editable: false,
+                            profiles: [],
+                        }
+                        builtInGroups.push(group)
+                    }
+
+                    group.profiles!.push(profile)
+                }
+
+                ungrouped.profiles = profiles.filter(p => !p.isBuiltin)
+            }
+
+            groups = groups.concat(builtInGroups)
+            groups.push(ungrouped)
+        }
+
+        return groups
+    }
+
+    /**
+    * Insert a new ProfileGroup in config
+    * arg: saveConfig (default: true) -> invoke after the Profile was updated
+    * arg: genId (default: true) -> generate uuid in before pushing Profile into config
+    */
+    async newProfileGroup (group: PartialProfileGroup<ProfileGroup>, saveConfig = true, genId = true): Promise<void> {
```

Comment:
same here

Reply:
Resolved in 5e5c80832de2d2125a9e55d89321071365c32938